### PR TITLE
feat(gate,cli): `$DENO_V8_FLAGS`

### PR DESCRIPTION
**Pull request by Yohe-Am**
Enables the `DENO_V8_FLAGS` env var for tuning v8.

#### Motivation and context

MET-435 or #621 

#### Migration notes

- This just exposes the deno paramter directly. Refer to deno or v8 docs for more details.

### Checklist

- [x] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [x] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by Natoandro — COMMENTED** [review]
Is it possible to add some test?
Can you update the doc?

**Comment by Yohe-Am** [comment]
@Natoandro done. Can you verify the env var descriptions I've added?

File: website/docs/reference/typegate/index.mdx [hunk 1]
```diff
@@ -46,3 +46,32 @@ GraphQL, being a query language, offers a great asset for Metatype's philosophy:
...
+| JWT_MAX_DURATION_SEC      | The lifetime of generated JWT access tokens.                     | `30 * 24 * 3600`        |
+| JWT_REFRESH_DURATION_SEC  | The lifetime of generated JWT refresh tokens.                    | `5 * 60`                |
+| SENTRY_DSN                |                                                                  |                         |
+| SENTRY_SAMPLE_RATE        |                                                                  | 1                       |
+| SENTRY_TRACES_SAMPLE_RATE |                                                                  | 1                       |
+| TRUST_PROXY               | Weather to accept proxy headers when resolving request contexts. | false                   |
+| TRUST_HEADER_IP           | The header key on which to resolve request origin addresses.     | X-Forwarded-For         |
+| DENO_V8_FLAGS             | Flags for tuning the v8 javascript engine.                       |                         |
+
+
```
Comment (Natoandro) [inline website/docs/reference/typegate/index.mdx:76]: We have more [here](https://github.com/metatypedev/metatype/blob/1c6c8afeb6bebc44b672ef61efae40a58689f8b2/website/docs/reference/typegate/synchronization/index.mdx?plain=1#L52).
Perhaps a link or re-add them?

File: website/shared/install/typegate.mdx [hunk 2]
```diff
@@ -8,7 +8,7 @@ This is the easiest way to get started, yet it's not publicly accessible. You ca
 
 ### With Docker
 
-Install [Docker](https://docs.docker.com/get-docker/) and use the following `compose.yml` to launch a typegate node. Redis is the single required dependency, however in practice you will want to add database or other systems that the typegate can connect to.
+Install [Docker](https://docs.docker.com/get-docker/) and use the following `compose.yml` to launch a typegate node. For production workloads, Redis is a recommended dependency to be used as a persistence layer but the typegate will run using in-memory stores if no `REDIS_URL` env var is detected. In practice you might also want to add a database or other systems that the typegate can connect to.
```
Comment (Natoandro) [inline website/shared/install/typegate.mdx:11]: See: https://github.com/metatypedev/metatype/blob/1c6c8afeb6bebc44b672ef61efae40a58689f8b2/website/docs/reference/typegate/synchronization/index.mdx

**Review by Natoandro — CHANGES_REQUESTED** [review]
Great.
Needs to update the notes on `REDIS_URL`.

File: website/docs/reference/typegate/index.mdx [hunk 3]
```diff
@@ -46,3 +46,32 @@ GraphQL, being a query language, offers a great asset for Metatype's philosophy:
...
+| TIMER_POLICY_EVAL_RETRIES |                                                                  | 1                       |
+| JWT_MAX_DURATION_SEC      | The lifetime of generated JWT access tokens.                     | `30 * 24 * 3600`        |
+| JWT_REFRESH_DURATION_SEC  | The lifetime of generated JWT refresh tokens.                    | `5 * 60`                |
+| SENTRY_DSN                |                                                                  |                         |
+| SENTRY_SAMPLE_RATE        |                                                                  | 1                       |
+| SENTRY_TRACES_SAMPLE_RATE |                                                                  | 1                       |
+| TRUST_PROXY               | Weather to accept proxy headers when resolving request contexts. | false                   |
+| TRUST_HEADER_IP           | The header key on which to resolve request origin addresses.     | X-Forwarded-For         |
+| DENO_V8_FLAGS             | Flags for tuning the v8 javascript engine.                       |                         |
```
Comment (Natoandro) [inline website/docs/reference/typegate/index.mdx:74]: Can we add a link to the reference for v8?

File: dev/Dockerfile [hunk 4]
```diff
@@ -69,13 +69,13 @@ RUN cargo install wasm-tools@$WASM_TOOLS_VERSION --locked --debug
...
 RUN . $GHJK_ENV \
-    && cargo chef cook --recipe-path recipe.json --release --package typegate \
+    && cargo chef cook --recipe-path recipe.json --package typegate \
     && rm recipe.json
 
 COPY . .
 
 RUN . $GHJK_ENV \
-    && cargo build --release --package typegate
+    && cargo build --package typegate
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply dev/Dockerfile:78]: Oh god. Good catch, that must be what's breaking the CI.

File: website/docs/reference/typegate/index.mdx (hunk 3 above)
Comment (Yohe-Am) [reply website/docs/reference/typegate/index.mdx]: They don't really have an official documentation page for this but if you pass the `--help` flag, it'll output a description of every flag and exit. I'll add that to our docs.

File: website/shared/install/typegate.mdx [hunk 5]
```diff
@@ -8,7 +8,7 @@ This is the easiest way to get started, yet it's not publicly accessible. You ca
 
 ### With Docker
 
-Install [Docker](https://docs.docker.com/get-docker/) and use the following `compose.yml` to launch a typegate node. Redis is the single required dependency, however in practice you will want to add database or other systems that the typegate can connect to.
+Install [Docker](https://docs.docker.com/get-docker/) and use the following `compose.yml` to launch a typegate node. For mutli-instance production workloads, Redis and an S3 object store provider are required but the typegate will run using in-memory stores if no `SYNC_*` environment variable is detected. More details can be found [here](/docs/reference/typegate/synchronization). In practice you might also want to add a database or other systems that the typegate can connect to.
```
Comment (Natoandro) [inline website/shared/install/typegate.mdx:11]: Multi?

**Review by Natoandro — APPROVED** [review]
Great job.
Found one last type.

File: website/shared/install/typegate.mdx (hunk 5 above)
Comment (Yohe-Am) [reply website/shared/install/typegate.mdx]: Nice, thanks.

**Review by Natoandro — APPROVED** [review]

**Review by Natoandro — APPROVED** [review]

**Review by afmika — APPROVED** [review]

File: website/docs/reference/typegate/index.mdx [hunk 6]
```diff
@@ -46,3 +46,32 @@ GraphQL, being a query language, offers a great asset for Metatype's philosophy:
...
+| ------------------------- | ---------------------------------------------------------------- | ----------------------- |
+| HOSTNAME                  | Hostname that typegate is deployed on.                           | `getHostname()` result. |
+| TG_PORT                   | Tcp port to serve typegate APIs at.                              | 7890                    |
+| TG_ADMIN_PASSWORD         | Password use by the CLI/SDK to configure the typegate.           | **Required**            |
+| TG_SECRET                 | Symmetric key used to encrypt cookies and other things.          | **Required**.           |
+| TMP_DIR                   | Top-level temporary directory.                                   | `$PWD/tmp`              |
+| DEBUG                     | Enable debug output and other development paths.                 | false                   |
+| TIMER_MAX_TIMEOUT_MS      | Timeout for custom runtime functions and other proccesses.       | 3000                    |
+| TIMER_DESTROY_RESOURCES   |                                                                  | true                    |
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply website/docs/reference/typegate/index.mdx]: We'll add these in the docs rework ig. Will merge this.

**Pull request merged**